### DISCUSSION TITLE
zstash a whole ensemble or individual members?

### DISCUSSION THREAD

**Discussion by wagmanbe** [E3SM-Project/zstash#204] [Questions] [answered]
The [e3sm step-by-step guide](https://acme-climate.atlassian.net/wiki/spaces/ED/pages/2309226536/Running+E3SM+step-by-step+guide#2.b.ii.-Transfer-to-NERSC) demonstrates how to zstash simulations separately using a loop. The DECKv1 members are archived separately, consistent with the example. 

Why not archive ensemble members together? Is there a reason not to do that? I have 250 ensemble members (34 TB).  If I wanted to extract all 250, or a subset of files from all 250, wouldn't it be easier if they were archived together?

**Comment by forsyth2** [comment]
Hi @wagmanbe,

First I want to note that if you're using the latest Unified environment you should be using the [updated instructions](https://acme-climate.atlassian.net/wiki/spaces/ED/pages/2309226536/Running+E3SM+step-by-step+guide#2.a.-E3SM-Unified-v1.6.0-/-zstash-v1.2.0-or-greater). 

I'm not quite sure I understand the question. Are you asking why the HPSS path is `//nersc/home/<first letter>/<username>/E3SMv2/${EXP}` rather than say `//nersc/home/<first letter>/<username>/E3SMv2/<ensemble-name>/${EXP}`?

**Comment by golaz** [comment, marked as answer]
@wagmanbe,

If you have all your simulations in a common directory, you could certainly try to archive all of them at once. I don't think there is anything within zstash that would prevent from you doing so.

There are pros and cons to both approaches.

- Pro: simplicity of dealing with only a single zstash archive.
- Con: possibly reduced performance due to more sequential work.
